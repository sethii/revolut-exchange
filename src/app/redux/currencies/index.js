export * from './currencies.reducer';
export * from './currencies.actions';
