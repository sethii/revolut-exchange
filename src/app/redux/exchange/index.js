export * from './exchange.reducer';
export * from './exchange.actions';
