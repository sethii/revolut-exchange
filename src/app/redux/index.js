export * from './app';
export * from './currencies';
export * from './wallets';
export * from './exchange';
