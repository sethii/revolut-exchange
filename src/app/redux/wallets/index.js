export * from './wallets.reducer';
export * from './wallets.actions';
