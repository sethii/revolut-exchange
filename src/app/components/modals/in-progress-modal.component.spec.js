import React from 'react';
import renderer from 'react-test-renderer';
import { toClass } from 'recompose';
import { mount } from 'enzyme';

import { exchangeCurrencyFinished } from '../../redux';
import { InProgressModal } from './in-progress-modal.component';

describe('In progress modal component', () => {
  it('renders in progress modal', () => {
    const InProgressModalComponent = toClass(InProgressModal);
    const fakeStore = {
      getState: () => {},
      dispatch: () => jest.fn(),
      subscribe: () => jest.fn()
    };

    expect(renderer.create(<InProgressModalComponent display={true} store={fakeStore}/>).toJSON()).toMatchSnapshot();
  });

  it('renders null when display false passed', () => {
    const InProgressModalComponent = toClass(InProgressModal);
    const fakeStore = {
      getState: () => {},
      dispatch: () => jest.fn(),
      subscribe: () => jest.fn()
    };

    expect(renderer.create(<InProgressModalComponent display={false} store={fakeStore}/>).toJSON()).toMatchSnapshot();
  });

  it('handles done click', () => {
    const InProgressModalComponent = toClass(InProgressModal);
    const fakeStore = {
      getState: () => {},
      dispatch: jest.fn(),
      subscribe: () => jest.fn()
    };

    const wrapper = mount(<InProgressModalComponent display={true} store={fakeStore}/>);

    wrapper.find('button').simulate('click');

    expect(fakeStore.dispatch).toBeCalledWith(exchangeCurrencyFinished());
  });
});
