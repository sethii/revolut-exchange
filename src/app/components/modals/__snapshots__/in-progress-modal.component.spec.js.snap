// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`In progress modal component renders in progress modal 1`] = `
.glamor-5 {
  border-radius: 10px;
  width: 500px;
  height: auto;
  padding: 20px;
  background-color: white;
  margin: 0 auto;
  font-size: 14px;
}

.glamor-1 {
  font-size: 24px;
}

.glamor-3 {
  border: none;
  font-size: 16px;
  background-color: #4169E1;
  color: white;
  padding: 15px 10px;
  width: 100%;
  border-radius: 25px;
}

.glamor-3:hover {
  cursor: pointer;
}

<div
  className="glamor-4 glamor-5"
>
  <h1
    className="glamor-0 glamor-1"
  >
    Exchange finished
  </h1>
  <button
    className="glamor-2 glamor-3"
    onClick={[Function]}
  >
    Close
  </button>
</div>
`;

exports[`In progress modal component renders null when display false passed 1`] = `null`;
